go/database/mpt/forest_test.go

```diff
@@ -1702,11 +1710,27 @@ func TestForest_ErrorsAreForwardedAndCollected(t *testing.T) {
 
 	prepareRootLookupFailure := func(m *mocks) {
 		m.branches.EXPECT().Get(rootId.Index()).Return(BranchNode{}, injectedError)
+		m.extensions.EXPECT().Flush()
+		m.extensions.EXPECT().Close()
+		m.branches.EXPECT().Flush()
+		m.branches.EXPECT().Close()
+		m.accounts.EXPECT().Flush()
+		m.accounts.EXPECT().Close()
+		m.values.EXPECT().Flush()
+		m.values.EXPECT().Close()
```

Comment:
I do not think adding `Flush` and `Close` expectations is the right way. Instead, the exchanged stock implementations opened by the file should be restored before closing it.

Reply:
the mocked stocks are passed as a parameter to the factory func [here](https://github.com/Fantom-foundation/Carmen/blob/c32079bdf46b565720dcab534a17ec5f92559e2e/go/database/mpt/forest_test.go#L1848), so I think this is correct. Calling `forest.Close()` in this func resolved a huge leak found by your tool

Reply:
Right, did not check the full function. You are right.